freqtrade/plugins/pairlist/MarketCapPairList.py

```diff
@@ -153,11 +165,11 @@ def filter_pairlist(self, pairlist: List[str], tickers: Dict) -> List[str]:
             if market == "futures":
                 pair_format += f":{self._stake_currency.upper()}"
 
-            top_marketcap = marketcap_list[: self._max_rank :]
+            top_marketcap = marketcap_list[: self._max_rank:]
 
             for mc_pair in top_marketcap:
                 test_pair = f"{mc_pair.upper()}/{pair_format}"
-                if test_pair in pairlist:
+                if test_pair in pairlist and test_pair not in filtered_pairlist:
```

Comment:
why would we assume `tot_marketcap` is not an unique list but includes duplicates? (the only case this 2nd condition makes sense) ?

Reply:
yeah when I was testing it there have been some duplicates appearing in the filtered_pairlist 

not quite sure why

Reply:
yep if I use the layer-1 category I got  KLAY/USDT 2 times in 

![Screenshot from 2024-09-17 22-45-59](https://github.com/user-attachments/assets/86050a91-6dcd-4d48-ac02-1435a0eaee82)

["BTC/USDT","ETH/USDT","BNB/USDT","SOL/USDT","TON/USDT","TRX/USDT","ADA/USDT","AVAX/USDT","BCH/USDT","NEAR/USDT","SUI/USDT","APT/USDT","FIL/USDT","INJ/USDT","HBAR/USDT","ATOM/USDT","FTM/USDT","AR/USDT","ALGO/USDT","SEI/USDT","OM/USDT","FLOW/USDT","KLAY/USDT","EGLD/USDT","GALA/USDT","XTZ/USDT","XEC/USDT","CHZ/USDT","ASTR/USDT","ROSE/USDT","GNO/USDT","KAVA/USDT","ELF/USDT","ENJ/USDT","SAGA/USDT","DCR/USDT","CHR/USDT","KDA/USDT","GLMR/USDT","ONE/USDT","VANRY/USDT","WAXP/USDT","DUSK/USDT","MOVR/USDT","SYS/USDT","OMNI/USDT","HIVE/USDT","LTO/USDT","VIC/USDT","WAN/USDT","BEL/USDT","AMB/USDT","FIO/USDT","KLAY/USDT"]

Reply:
it also apears to be  twice in the get_coins_market response 

['btc', 'eth', 'bnb', 'sol', 'ton', 'trx', 'ada', 'avax', 'bch', 'near', 'kas', 'xmr', 'sui', 'apt', 'cro', 'fil', 'inj', 'hbar', 'atom', 'ftm', 'ar', 'algo', 'sei', 'om', 'core', 'flow', 'klay', 'egld', 'flr', 'gala', 'xtz', 'xec', 'ron', 'chz', 'astr', 'aioz', 'rose', 'gno', 'wemix', 'kava', 'elf', 'enj', 'xrd', 'zeta', 'qubic', 'saga', 'dcr', 'chr', 'kda', 'alph', 'glmr', 'one', 'xch', 'cspr', 'vanry', 'azero', 'ozo', 'waxp', 'dusk', 'orai', 'movr', 'sys', 'omni', 'hive', 'oas', 'zano', 'tet', 'cudos', 'lyx', 'qanx', 'lto', 'erg', 'octa', 'kuji', 'meld', 'heart', 'trias', 'vic', 'ccd', 'wan', '$smh', 'bld', 'synt', 'dnx', 'bel', 'stos', 'tara', 'amb', 'glq', 'fio', 'rxd', 'dmd', 'htr', 'jkl', 'arch', 'dvpn', 'nexa', 'saito', 'koin', 'evmos', 'vara', 'xep', 'canto', 'xel', 'ngl', 'xna', 'epic', 'fuse', 'pica', 'soul', 'tht', 'caps', 'aura', 'nrg', 'blocx', 'kex', 'hmnd', 'mas', 'wit', 'nai', 'xcb', 'huahua', 'pkt', 'scpt', 'netz', 'btsg', 'libra', 'cph', 'hac', 'mtv', 'whale', 'intr', 'kls', 'c4e', 'unq', 'kint', 'aura', 'spr', 'dsm', 'met', 'eqpay', 'hoot', 'ixi', 'kcn', 'dec', 'lila', 'scr', 'lcn', 'busy', 'bit', 'root', 'pyi', 'sge', 'mpc', 'ethg', 'coreum', 'nibi', 'p3d', 'cirx', '$muu', 'drc', 'ulx', 'water', 'l1x', 'wchz', 'kalis', 'php', 'klay', 'minima', '']

Reply:
ah, the naming collision problem :laughing:  makes sense, then.

"symbol" in coingecko is not necessarily unique - KLAY will map to [Klaytn](https://www.coingecko.com/en/coins/klaytn) and [kaia](https://www.coingecko.com/en/coins/kaia).

happens with tons of coins, really (there's a mapping for known affected coins in `fiat_convert.py`) - the problem is, you can't reliably know which one is the one trade on exchange (it's usually the one higher up on lists).